parserator_web/views.py

```diff
@@ -16,9 +15,30 @@ class AddressParse(APIView):
     def get(self, request):
         # TODO: Flesh out this method to parse an address string using the
         # parse() method and return the parsed components to the frontend.
-        return Response({})
+
+        address_components, address_type = {}, ''
+
+        # in case the this endpoint is ever called without the
+        # address parameter
+        if 'address' not in request.query_params:
+            return Response({'input_string': '',
+                             'address_components': address_components,
+                             'address_type': address_type})
+
+        input_string = request.query_params['address']
+
+        try:
+            address_components, address_type = self.parse(input_string)
+        except usaddress.RepeatedLabelError as e:
+            address_components, address_type = e.parsed_string, 'Error'
+        finally:
+            return Response({'input_string': input_string,
+                             'address_components': address_components,
+                             'address_type': address_type})
 
     def parse(self, address):
         # TODO: Implement this method to return the parsed components of a
         # given address using usaddress: https://github.com/datamade/usaddress
-        return address_components, address_type
+
+        address_components, address_type = usaddress.tag(address)
+        return (address_components, address_type)
```

Comment:
You can tighten this up by returning `usaddress.tag(address)` directly.

Reply:
Agreed.